packages/react-pixi/src/reconciler/hostconfig.js

```diff
@@ -28,27 +28,43 @@ function appendChild(parent, child)
     }
 }
 
-function removeChild(parent, child)
+function willUnmountRecursive(child, parent)
 {
-    if (typeof child.willUnmount === 'function')
-    {
-        child.willUnmount(child, parent);
-    }
+    child.willUnmount?.(child, parent);
 
-    // unmount potential children
-    if (child?.config?.destroyChildren !== false && child.children?.length)
+    // ensure willUnmount is called on children, but don't actually destroy them
+    if (child.config?.destroyChildren !== false && child.children?.length)
     {
         [...child.children].forEach((c) =>
         {
-            removeChild(child, c);
+            // TODO: should we call willUnmount anyway irrespective of whether destroyChildren is true?
+            willUnmountRecursive(c, child);
```

Comment:
@Zyie I'm wondering whether we should just call `willUnmount` on children anyway whether or not `destroyChildren` is true - technically the child is about to unmount irrespective of whether it's being destroyed